### DISCUSSION TITLE
can't use cv2 after pip install albumentations on azure vm

### DISCUSSION THREAD

**Discussion by firasam** [EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge#51] [3.0 VMs and data (Azure VM, local deployment, and Jupyter)]
hello all. recently i 
installed albumentations (https://pypi.org/project/albumentations/#installation)
using pip install comand on jupyter notebook on azure vm. and after i install it i cant use cv2 because if i import cv2 it will throw the following error:""ImportError: libXrender.so.1: cannot open shared object file: No such file or directory

judging by this link 
https://albumentations.ai/docs/faq/#i-successfully-installed-the-library-but-when-i-am-trying-to-import-it-i-receive-an-error-importerror-libxrenderso1-cannot-open-shared-object-file-no-such-file-or-directory

seems like the solution is to do sudo apt-get update to update the cv2 and the depedencies. but i can't do it on  the jupyter docker on the azure vm. is there any way to fix this?

**Comment by EY-glowglowglow** [comment]
@codeindulgence  any ideas? :)

**Reply by cheukhin1024** [reply]
> hello all. recently i
> installed albumentations (https://pypi.org/project/albumentations/#installation)
> using pip install comand on jupyter notebook on azure vm. and after i install it i cant use cv2 because if i import cv2 it will throw the following error:""ImportError: libXrender.so.1: cannot open shared object file: No such file or directory
> 
> judging by this link
> https://albumentations.ai/docs/faq/#i-successfully-installed-the-library-but-when-i-am-trying-to-import-it-i-receive-an-error-importerror-libxrenderso1-cannot-open-shared-object-file-no-such-file-or-directory
> 
> seems like the solution is to do sudo apt-get update to update the cv2 and the depedencies. but i can't do it on the jupyter docker on the azure vm. is there any way to fix this?


Try this SimpleCV.
[]http://simplecv.org/

**Comment by nickbutler** [comment]
Hi @firasam,

In order to use `apt` you'll need to access the container shell as `root`.
First [access the VM terminal](https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/4a.-Troubleshooting:-FAQ#access-the-vm-terminal), from there you can access the Jupyter container as root:

```
cd /opt/odc
sudo docker-compose exec -u root jupyter bash
```

From there you can `apt-get install ...` to your hearts content!
